pptx/dml/line.py

```diff
@@ -165,3 +251,162 @@ def width(self):
+    """
+
+    def __init__(self, eg_join_properties_parent, join_obj):
+        super(JoinFormat, self).__init__()
+        self._xPr = eg_join_properties_parent
+        self._join = join_obj
+
+    @classmethod
+    def from_join_parent(cls, eg_joinProperties_parent):
+        """
+        Return a |JoinFormat| instance initialized to the settings contained
+        in *eg_joinProperties_parent*, which must be an emelent having 
+        eg_lineJoinProperties in its child element sequence in the XML schemea.
+        """
+        join_elm = eg_joinProperties_parent.eg_lineJoinProperties
+        join = _Join(join_elm)
+        join_format = cls(eg_joinProperties_parent, join)
+        return join_format
+
+    def round(self):
+        """
+        Sets the join type to Round
+        """
+        roundJoin = self._xPr.get_or_change_to_round()
+        self._join = _RoundJoin(roundJoin)
+
+    def bevel(self):
+        """
+        Sets the join type to Bevel
+        """
+        bevelJoin = self._xPr.get_or_change_to_bevel()
+        self._join = _BevelJoin(bevelJoin)
+
+    def miter(self):
+        """
+        Sets the join type to Miter
+        """
+        miterJoin = self._xPr.get_or_change_to_miter()
+        self._join = _MiterJoin(miterJoin)
+
+    @property
+    def limit(self):
+        """
+        The setting for the limit on a Miter Join. 
+        """
+        return self._join.limit
+
+    @limit.setter
+    def limit(self, val):
+        join = self._xPr.get_or_change_to_miter()
+        self._join = _MiterJoin(join)
+        self._join.limit = val
+
+    @property
+    def type(self):
+        """
+        Returns the type value string
+        """
+        return self._join.type
+    
+
+class _Join(object):
+    """
+    Object factor for join object of class matching join element.
+    """
+
+    def __new__(cls, xJoin):
+        if xJoin is None:
+            join_cls = _NoneJoin
+        elif isinstance(xJoin, CT_LineJoinRound):
+            join_cls = _RoundJoin
+        elif isinstance(xJoin, CT_LineJoinBevel):
+            join_cls = _BevelJoin
+        elif isinstance(xJoin, CT_LineJoinMiterProperties):
+            join_cls = _MiterJoin
+        else:
+            join_cls = _Join
+        return super(_Join, cls).__new__(join_cls)
+
+    
+class _NoneJoin(_Join):
+    @property
+    def type(self):
+        return None
+
+class _RoundJoin(_Join):
+    @property
+    def type(self):
+        return "round"
+
+class _BevelJoin(_Join):
+    @property
+    def type(self):
+        return "bevel"
+
+class _MiterJoin(_Join):
+    def __init__(self, miterJoin):
+        self._element = miterJoin
+
+    @property
+    def type(self):
+        return "miter"
+    
+    @property
+    def limit(self):
+        """
+        Property defining the angle of the miter.  Must be a positive percentage
+        """
+        return self._element.lim
+
+    @limit.setter
+    def limit(self, val):
+        self._element.lim = val
```

Comment:
Based on the doc string on line 367 limit shouldn't be negative, so I'm thinking that we should add a check for negative values, or are we not setting this from the client side?

Reply:
This happens automatically by the validation as part of the simple type class

Reply:
I left another comment where this is happening

Reply:
Thank you!